packages/cli/src/services/annotationTag.service.ts

```diff
@@ -0,0 +1,86 @@
+import { Service } from 'typedi';
+import { validateEntity } from '@/GenericHelpers';
+import type { ITagWithCountDb } from '@/Interfaces';
+import type { AnnotationTagEntity } from '@db/entities/AnnotationTagEntity';
+import { AnnotationTagRepository } from '@db/repositories/annotationTag.repository';
+import { ExternalHooks } from '@/ExternalHooks';
+
+type GetAllResult<T> = T extends { withUsageCount: true }
+	? ITagWithCountDb[]
+	: AnnotationTagEntity[];
+
+@Service()
+export class AnnotationTagService {
+	constructor(
+		private externalHooks: ExternalHooks,
+		private tagRepository: AnnotationTagRepository,
+	) {}
+
+	toEntity(attrs: { name: string; id?: string }) {
+		attrs.name = attrs.name.trim();
+
+		return this.tagRepository.create(attrs);
+	}
+
+	async save(tag: AnnotationTagEntity, actionKind: 'create' | 'update') {
+		await validateEntity(tag);
+
+		const action = actionKind[0].toUpperCase() + actionKind.slice(1);
+
+		await this.externalHooks.run(`annotationTag.before${action}`, [tag]);
+
+		const savedTag = this.tagRepository.save(tag, { transaction: false });
+
+		await this.externalHooks.run(`annotationTag.after${action}`, [tag]);
+
+		return await savedTag;
```

Comment:
I believe the `await` should be in the `.save` call. I would assume that `afterCreate` hook is not called unless the entity already exists in DB